users/views.py

```diff
@@ -0,0 +1,188 @@
+        except IntegrityError as integrity_error:
+            response["message"] = str(integrity_error)
+            response["code"] = status.HTTP_409_CONFLICT
+        except ValidationError as validation_error:
+            response["message"] = " ".join(validation_error)
+            response["code"] = status.HTTP_400_BAD_REQUEST
+
+        return Response(response)
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def patch(self, request):
+        """
+        This function updates data of existing user
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        try:
+            user = request.user
+            if "email" in request.POST:
+                email = request.POST["email"]
+                validate_email(email)
+                if not User.objects.filter(email=email).exists():
+                    user.email = request.POST["email"]
+            if "name" in request.POST:
+                name = request.POST["name"]
+                if name not in ["", " "]:
+                    user.name = name
+                else:
+                    return Response(
+                        {
+                            "message": "Name is not valid",
+                            "code": status.HTTP_400_BAD_REQUEST,
+                        }
+                    )
+            if "password" in request.POST:
+                validate_password(request.POST["password"])
+                user.password = make_password(request.POST["password"])
+            user.save()
+            return Response(
+                {"message": "Changes updated successfully", "code": status.HTTP_200_OK}
+            )
+        except ValidationError as error:
+            return Response(
+                {"message": " ".join(error), "code": status.HTTP_400_BAD_REQUEST}
+            )
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def delete(self, request):
+        """
+        This function deletes a user from database
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        user = request.user
+        response = {}
+        if user.is_superuser:
+            try:
+                get_object_or_404(User, email=request.POST["email"]).delete()
+                response["message"] = "User deleted successfully"
+                response["code"] = status.HTTP_200_OK
+            except Http404:
+                response["message"] = "No user exists with given email"
+                response["code"] = status.HTTP_400_BAD_REQUEST
+            except KeyError:
+                response["message"] = "No email provided"
+                response["code"] = status.HTTP_400_BAD_REQUEST
```

Comment:
Does every `KeyError` mean no email provided?

Reply:
No, but in this case it does as the only dictionary access is `request.POST["email"]`. If a `KeyError` arises then it means no email was provided in the request

Reply:
Let's place it in the scenario of the future since we are using `No email provided` as a generic response to a generic exception `KeyError`.  Someone adds another key(`id`) at some point and misses to see this except block. what message will be thrown if the key `Id` is missing now?